src/Microsoft.ML.Data/Transforms/CopyColumnsTransform.cs

```diff
@@ -163,4 +169,189 @@ protected override Delegate GetGetterCore(IChannel ch, IRow input, int iinfo, ou
             return (Delegate)methodInfo.Invoke(input, new object[] { col });
         }
     }
+
+    public sealed class CopyColumnsEstimator : IEstimator<CopyColumnsTransformer>
+    {
+        private readonly (string Source, string Name)[] _columns;
+        private readonly IHost _host;
+
+        public CopyColumnsEstimator(IHostEnvironment env, string input, string output)
+        {
+            Contracts.CheckNonWhiteSpace(input, nameof(input));
+            Contracts.CheckNonWhiteSpace(output, nameof(output));
+            _columns = new (string, string)[1] { (input, output) };
+            _host = env.Register("CopyColumnsEstimator");
+        }
+
+        public CopyColumnsEstimator(IHostEnvironment env, (string Source, string Name)[] columns)
+        {
+            Contracts.CheckValue(columns, nameof(columns));
+            var newNames = new HashSet<string>();
+            foreach (var column in columns)
+            {
+                if (newNames.Contains(column.Name))
+                    throw Contracts.ExceptUserArg(nameof(columns), $"New column {column.Name} specified multiple times");
+                newNames.Add(column.Name);
+            }
+            _columns = columns;
```

Comment:
`columns.Select(x=>x.Name).Distinct().Count`

Reply:
I would prefer to tell user where he make mistake.

---
In reply to: [211790617](https://github.com/dotnet/machinelearning/pull/706#discussion_r211790617) [](ancestors = 211790617)